src/analysis/seminorm.lean

```diff
@@ -348,6 +388,29 @@ begin
   ...    < r   : by rwa mem_ball_zero at hy,
 end
 
+lemma ball_finset_sup_eq_Inter (p : ι → seminorm 𝕜 E) (s : finset ι) (e : E) {r : ℝ} (hr : 0 < r) :
+  ball (s.sup p) e r = ⋂ (i ∈ s), ball (p i) e r :=
+begin
+  lift r to nnreal using hr.le,
+  simp_rw [ball, Inter_set_of, finset_sup_apply, nnreal.coe_lt_coe,
+    finset.sup_lt_iff (show ⊥ < r, from hr), ←nnreal.coe_lt_coe, subtype.coe_mk],
+end
+
+lemma ball_sup (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} (hr : 0 < r) :
+  ball (p ⊔ q) e r = (ball p e r) ⊓ (ball q e r) :=
+begin
+  lift r to nnreal using hr.le,
+  simp_rw [set.inf_eq_inter, ball, ←set.set_of_and],
+  simp,
+end
+
```

Comment:
I think this is the statement of the last lemma that @YaelDillies and I would like in this PR:
```suggestion

@[simp] lemma ball_zero' (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} (hr : 0 < r) :
  ball (0 : seminorm 𝕜 E) e r = set.univ :=
sorry

```

Reply:
Sorry, didn't see that either. Should I also make the version for `ball_bot`?

Reply:
It's probably a good idea so that `simp` can be more clever and the user doesn't have to find `zero_eq_bot`.